platform/rpc/nats/bus.go

```diff
@@ -0,0 +1,246 @@
+// useful for synchronising during testing.
+func (n *NATS) AwaitPresence(instID flux.InstanceID, timeout time.Duration) error {
+	timer := time.After(timeout)
+	attempts := time.NewTicker(presenceTick)
+	defer attempts.Stop()
+
+	for {
+		select {
+		case <-attempts.C:
+			if err := n.Ping(instID); err == nil {
+				return nil
+			}
+		case <-timer:
+			return errors.New("presence timeout")
+		}
+	}
+}
+
+func (n *NATS) Ping(instID flux.InstanceID) error {
+	var response PingResponse
+	err := n.snd.Request(string(instID)+methodPing, ping{}, &response, timeout)
+	if err == nil {
+		err = maybeError(response.Error)
+	}
+	return err
+}
+
+type AllServicesResponse struct {
+	Services []platform.Service
+	Error    string
+}
+
+type SomeServicesResponse struct {
+	Services []platform.Service
+	Error    string
+}
+
+type RegradeResponse struct {
+	Result rpc.RegradeResult
+	Error  string
+}
+
+type ping struct{}
+
+type PingResponse struct {
+	Error string
+}
+
+func maybeError(msg string) error {
+	if msg != "" {
+		return errors.New(msg)
+	}
+	return nil
+}
+
+func maybeString(err error) string {
+	if err != nil {
+		return err.Error()
+	}
+	return ""
+}
+
+// requester just collect the things you need to make a request
+// together
+type requester struct {
+	conn     *nats.EncodedConn
+	instance string
+}
+
+func (r *requester) AllServices(ns string, ig flux.ServiceIDSet) ([]platform.Service, error) {
+	var response AllServicesResponse
+	if err := r.conn.Request(r.instance+methodAllServices, rpc.AllServicesRequest{ns, ig}, &response, timeout); err != nil {
+		return nil, err
+	}
+	return response.Services, maybeError(response.Error)
+}
+
+func (r *requester) SomeServices(incl []flux.ServiceID) ([]platform.Service, error) {
+	var response SomeServicesResponse
+	if err := r.conn.Request(r.instance+methodSomeServices, incl, &response, timeout); err != nil {
+		return nil, err
+	}
+	return response.Services, maybeError(response.Error)
+}
+
+func (r *requester) Regrade(specs []platform.RegradeSpec) error {
+	var response RegradeResponse
+	if err := r.conn.Request(r.instance+methodRegrade, specs, &response, timeout); err != nil {
+		return err
+	}
+	if len(response.Result) > 0 {
+		errs := platform.RegradeError{}
+		for s, e := range response.Result {
+			errs[s] = errors.New(e)
+		}
+		return errs
+	}
+	return maybeError(response.Error)
+}
+
+func (r *requester) Ping() error {
+	var response PingResponse
+	if err := r.conn.Request(r.instance+methodPing, ping{}, &response, timeout); err != nil {
+		return err
+	}
+	return maybeError(response.Error)
+}
+
+// Connect returns a platform.Platform implementation that can be used
+// to talk to a particular instance.
+func (n *NATS) Connect(instID flux.InstanceID) (platform.Platform, error) {
+	return &requester{
+		conn:     n.snd,
+		instance: string(instID),
+	}, nil
+}
+
+// Subscribe registers a remote platform.Platform implementation as
+// representing a particular instance ID, blocking indefinitely.
```

Comment:
Nope it's not accurate, it was carried across from an older comment on `platform/standalone.go`. Ops! I'll mend it.